source/binding/OperatorExpressions.cpp

```diff
@@ -1498,6 +1501,145 @@ void ReplicationExpression::serializeTo(ASTSerializer& serializer) const {
     serializer.write("concat", concat());
 }
 
+Expression& StreamingConcatenationExpression::fromSyntax(
+    Compilation& compilation, const StreamingConcatenationExpressionSyntax& syntax,
+    const BindContext& context, const Type* assignmentTarget) {
+    const bool isRightToLeft = syntax.operatorToken.kind == TokenKind::LeftShift;
+    size_t sliceSize = 0;
+
+    auto badResult = [&]() {
+        return compilation.emplace<StreamingConcatenationExpression>(
+            compilation.getErrorType(), sliceSize, span<const Expression*>(), syntax.sourceRange());
+    };
+
+    if (!(context.flags & BindFlags::StreamingAllowed)) {
+        context.addDiag(diag::BadStreamContext, syntax.operatorToken.location());
+        return badExpr(compilation, badResult());
+    }
+
+    if (syntax.sliceSize) {
+        if (isRightToLeft) {
+            const auto& sliceExpr =
+                bind(*syntax.sliceSize, context, BindFlags::AllowDataType | BindFlags::Constant);
+            if (sliceExpr.bad())
+                return badExpr(compilation, badResult());
+            if (sliceExpr.kind == ExpressionKind::DataType) {
+                if (!sliceExpr.type->isFixedSize()) {
+                    auto& diag = context.addDiag(diag::BadStreamSlice, sliceExpr.sourceRange);
+                    if (sliceExpr.type->location)
+                        diag.addNote(diag::NoteDeclarationHere, sliceExpr.type->location);
+                    return badExpr(compilation, badResult());
+                }
+                sliceSize = sliceExpr.type->bitstreamWidth();
+            }
+            else {
+                optional<int32_t> count = context.evalInteger(sliceExpr);
+                if (!count)
+                    return badExpr(compilation, badResult());
+                if (*count <= 0) {
+                    context.requireGtZero(count, sliceExpr.sourceRange);
```

Comment:
You can simply this by checking the result of the requireGtZero call in the if statement instead of checking the value yourself.

Reply:
Yes, this is a good idea. I simplify it into
```c++
                if (!context.requireGtZero(count, sliceExpr.sourceRange))
                    return badExpr(compilation, badResult());
```